src/commonMain/kotlin/dev/snipme/highlights/Highlights.kt

```diff
@@ -20,6 +26,9 @@ class Highlights private constructor(
     var snapshot: CodeSnapshot? = null
         private set
 
+    var analysisJob: Job? = null
```

Comment:
This assignment is not thread safe and can leak, I'd recommend to avoid storing a job. See my comment below for a suggestion.

Reply:
I see, this is probably to change